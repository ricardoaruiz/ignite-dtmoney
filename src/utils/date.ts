export const formatBRLDate = (date: Date) => {
  return Intl.DateTimeFormat('pt-BR').format(date);
};
