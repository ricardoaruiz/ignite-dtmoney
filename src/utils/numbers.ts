/**
 * Format value with currency pattern
 */
export const formatBrlCurrency = (value: number): string => {
  return Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(value);
};
