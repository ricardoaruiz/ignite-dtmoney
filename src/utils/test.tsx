import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/themes/theme';

export const renderWithTheme = (children: React.ReactNode) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
