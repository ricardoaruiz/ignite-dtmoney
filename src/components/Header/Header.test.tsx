/* eslint-disable react/display-name */
import { renderWithTheme } from 'utils/test';
import { Header } from './index';

jest.mock('components/NewTransactionModal', () => ({
  NewTransactionModal: () => <div></div>,
}));

describe('<Header />', () => {
  it('should be render correctly', () => {
    const onOpenNewTransactionModalMocked = jest.fn();
    const { container } = renderWithTheme(
      <Header onOpenNewTransactionModal={onOpenNewTransactionModalMocked} />,
    );

    expect(container.parentElement).toMatchSnapshot();
  });
});
