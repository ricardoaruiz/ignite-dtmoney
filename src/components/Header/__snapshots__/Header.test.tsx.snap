// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should be render correctly 1`] = `
<body>
  <div>
    <header
      class="sc-ksluID dEOKnc"
    >
      <div
        class="sc-bdnxRM ZSPUe"
      >
        <img
          alt="dt money"
          src="logo.svg"
        />
        <button
          type="button"
        >
          Nova transação
        </button>
      </div>
    </header>
  </div>
</body>
`;
